process/block/metablock_test.go

```diff
@@ -3061,3 +3061,442 @@ func TestMetaProcessor_CreateNewHeaderValsOK(t *testing.T) {
+				assert.True(t, wasCalled)
+				return nil
+			},
+		}
+
+		mp, _ := blproc.NewMetaProcessor(arguments)
+
+		body := &block.Body{
+			MiniBlocks: []*block.MiniBlock{
+				{
+					ReceiverShardID: 1,
+					SenderShardID:   1,
+				},
+			},
+		}
+
+		err := mp.ProcessEpochStartMetaBlock(header, body)
+		assert.Nil(t, err)
+	})
+}
+
+func TestMetaProcessor_UpdateEpochStartHeaderShouldFail(t *testing.T) {
+	t.Parallel()
+
+	accFeesInEpoch := big.NewInt(1000)
+	devFeesInEpoch := big.NewInt(100)
+	blkc, _ := blockchain.NewMetaChain(&statusHandlerMock.AppStatusHandlerStub{})
+	_ = blkc.SetCurrentBlockHeaderAndRootHash(
+		&block.MetaBlock{
+			Round:                  1,
+			Nonce:                  1,
+			AccumulatedFeesInEpoch: accFeesInEpoch,
+			DevFeesInEpoch:         devFeesInEpoch,
+		}, []byte("root hash"),
+	)
+	_ = blkc.SetGenesisHeader(&block.MetaBlock{Nonce: 0})
+
+	coreComponents, dataComponents, bootstrapComponents, statusComponents := createMockComponentHolders()
+	dataComponents.BlockChain = blkc
+	hash := []byte("hash1")
+
+	arguments := createMockMetaArguments(coreComponents, dataComponents, bootstrapComponents, statusComponents)
+
+	expectedErr := errors.New("expected error")
+	arguments.EpochEconomics = &mock.EpochEconomicsStub{
+		ComputeEndOfEpochEconomicsCalled: func(metaBlock *block.MetaBlock) (*block.Economics, error) {
+			return nil, expectedErr
+		},
+	}
+
+	mp, _ := blproc.NewMetaProcessor(arguments)
+
+	header := &block.MetaBlock{
+		Nonce:                  1,
+		Round:                  1,
+		PrevHash:               hash,
+		AccumulatedFeesInEpoch: big.NewInt(0),
+		DevFeesInEpoch:         big.NewInt(0),
+	}
+
+	err := mp.UpdateEpochStartHeader(header)
+	assert.Equal(t, expectedErr, err)
+}
+
+func TestMetaProcessor_UpdateEpochStartHeader(t *testing.T) {
+	t.Parallel()
+
+	accFeesInEpoch := big.NewInt(1000)
+	devFeesInEpoch := big.NewInt(100)
+	blkc, _ := blockchain.NewMetaChain(&statusHandlerMock.AppStatusHandlerStub{})
+	_ = blkc.SetCurrentBlockHeaderAndRootHash(
+		&block.MetaBlock{
+			Round:                  1,
+			Nonce:                  1,
+			AccumulatedFeesInEpoch: accFeesInEpoch,
+			DevFeesInEpoch:         devFeesInEpoch,
+		}, []byte("root hash"),
+	)
+	_ = blkc.SetGenesisHeader(&block.MetaBlock{Nonce: 0})
+	coreComponents, dataComponents, bootstrapComponents, statusComponents := createMockComponentHolders()
+	dataComponents.BlockChain = blkc
+	hash := []byte("hash1")
+
+	arguments := createMockMetaArguments(coreComponents, dataComponents, bootstrapComponents, statusComponents)
+
+	mp, _ := blproc.NewMetaProcessor(arguments)
+
+	header := &block.MetaBlock{
+		Nonce:                  1,
+		Round:                  1,
+		PrevHash:               hash,
+		AccumulatedFeesInEpoch: big.NewInt(0),
+		DevFeesInEpoch:         big.NewInt(0),
+	}
+
+	err := mp.UpdateEpochStartHeader(header)
+	assert.Nil(t, err)
+	assert.Equal(t, accFeesInEpoch, header.GetAccumulatedFeesInEpoch())
```

Comment:
can you also check end of epoch economics? 
you can set the ComputeEndOfEpochEconomicsCalled to return some predefined value, and have that checked on the header.GetEpochStartHandler().GetEconomicsHandler()